### PR TITLE
docs: restructure and clarify learning-rate.md documentation

### PR DESCRIPTION
**Pull request by OutisLi**
## Summary

This PR completely restructures the `learning-rate.md` documentation to improve clarity, organization, and accuracy. The previous version had content scattered across sections with significant repetition. The new structure follows a user-centric approach: quick start → configuration reference → mathematical theory.

### Structural Improvements
- **Reorganized section order**: Quick Start → Parameters → Schedule Types → Warmup → Mathematical Theory → Migration Guide
- **Eliminated content duplication**: Removed redundant formulas between Theory and Instructions sections

## Summary by CodeRabbit

* **Documentation**
  * Reworked learning-rate guide into a structured, example-driven reference with Quick Start, explicit exponential and cosine schedules, and JSON configuration examples.
  * Added a Notation/Theory section, clear warmup formulas and mutual‑exclusivity rules, unified parameter descriptions (start_lr/stop_lr/stop_lr_ratio/decay_steps) and smooth vs stepped behavior.
  * Expanded migration guidance for versions prior to 3.1.3 and refreshed references.

### CODE REVIEW

File: doc/train/learning-rate.md [hunk 1]
```diff
@@ -1,306 +1,363 @@
...
+    "start_lr": 0.001,
+    "stop_lr_ratio": 1e-3,
+    "decay_steps": 5000
+}
 ```
 
-By setting `smooth: true`, the learning rate decays smoothly at every step instead of in a stepped manner. This provides a more gradual decay curve similar to PyTorch's `ExponentialLR`, whereas the default stepped mode (`smooth: false`) is similar to PyTorch's `StepLR`.
-
-[^1]: This section is built upon Jinzhe Zeng, Duo Zhang, Denghui Lu, Pinghui Mo, Zeyu Li, Yixiao Chen, Marián Rynik, Li'ang Huang, Ziyao Li, Shaochen Shi, Yingze Wang, Haotian Ye, Ping Tuo, Jiabin Yang, Ye Ding, Yifan Li, Davide Tisi, Qiyu Zeng, Han Bao, Yu Xia, Jiameng Huang, Koki Muraoka, Yibo Wang, Junhan Chang, Fengbo Yuan, Sigbjørn Løland Bore, Chun Cai, Yinnian Lin, Bo Wang, Jiayan Xu, Jia-Xin Zhu, Chenxing Luo, Yuzhi Zhang, Rhys E. A. Goodall, Wenshuo Liang, Anurag Kumar Singh, Sikai Yao, Jingchao Zhang, Renata Wentzcovitch, Jiequn Han, Jie Liu, Weile Jia, Darrin M. York, Weinan E, Roberto Car, Linfeng Zhang, Han Wang, [J. Chem. Phys. 159, 054801 (2023)](https://doi.org/10.1063/5.0155600) licensed under a [Creative Commons Attribution (CC BY) license](http://creativecommons.org/licenses/by/4.0/).
+## References
```
Comment (njzjz) [inline doc/train/learning-rate.md:268]: This section is empty?

File: doc/train/learning-rate.md [hunk 2]
```diff
@@ -1,306 +1,363 @@
...
-```math
-    r = {\left(\frac{\gamma^{\text{stop}}}{\gamma^0}\right )} ^{\frac{s}{\tau^{\text{decay}}}},
-```
+You must specify exactly one of the following two mutually exclusive parameters:
 
-where $\tau^{\text{decay}} = \tau^{\text{stop}} - \tau^{\text{warmup}}$ is the number of decay steps, $\tau^{\text{stop}} \in \mathbb{N}$ is the total training steps, $\gamma^{\text{stop}} \in \mathbb{R}$ is the stopping learning rate, and $s \in \mathbb{N}$ is the decay steps.
+| Parameter       | Type  | Description                                                                                                              |
+| --------------- | ----- | ------------------------------------------------------------------------------------------------------------------------ |
+| `stop_lr`       | float | The learning rate at the end of training. When `decay_rate` is explicitly set, this serves as the minimum learning rate. |
+| `stop_lr_ratio` | float | The ratio of `stop_lr` to `start_lr`. Computed as `stop_lr = start_lr * stop_lr_ratio`.                                  |
```
Comment (njzjz) [inline doc/train/learning-rate.md]: These tables seem to be duplicate of the parameter doc. At least, they should be generated automatically - we should not maintain them.

File: doc/train/learning-rate.md [hunk 3]
```diff
@@ -1,306 +1,363 @@
...
+```
 
-#### Migration examples
+### Decay rate computation
 
-**Before (legacy configuration):**
+If `decay_rate` is not explicitly provided, it is computed from `start_lr` and `stop_lr`:
+
+```text
+decay_rate = (stop_lr / start_lr) ^ (decay_steps / (numb_steps - warmup_steps))
```
Comment (njzjz) [inline doc/train/learning-rate.md]: It seems to me that equations appear twice.

File: doc/train/learning-rate.md [hunk 4]
```diff
@@ -1,306 +1,363 @@
...
+These parameters are **mutually exclusive**:
 
-The following example shows cosine annealing with warmup.
+- `stop_lr` and `stop_lr_ratio`
+- `warmup_steps` and `warmup_ratio`
 
-**Example 7: Cosine annealing with warmup**
+### Migration examples
+
+**Before (legacy):**
```
Comment (njzjz) [inline doc/train/learning-rate.md]: It is unclear what migration and legacy mean in this doc.

File: doc/train/learning-rate.md [hunk 5]
```diff
@@ -1,306 +1,363 @@
...
-        "warmup_start_factor": 0.0
-    }
+"learning_rate": {
+    "type": "exp",
+    "decay_steps": 5000
+}
 ```
 
-In this example, the learning rate starts from `0.0` and increases linearly to `0.001` over the first 5,000 steps, then follows a cosine annealing curve down to `1e-6`.
+**After (updated):**
```
Comment (njzjz) [inline doc/train/learning-rate.md]: Also, it seems that "legacy" has the same parameters as "updated" without changes.

**Review by njzjz — CHANGES_REQUESTED** [review]

**Review by njzjz — COMMENTED** [review]
It seems to me all parameter links have been removed?

**Review by njzjz — APPROVED** [review]
